packages/client/bin/startRpc.ts

```diff
@@ -38,7 +38,13 @@ type RPCArgs = {
 function parseJwtSecret(config: Config, jwtFilePath?: string): Buffer {
   let jwtSecret: Buffer
   const defaultJwtPath = `${config.datadir}/jwtsecret`
-  if (jwtFilePath !== undefined || existsSync(defaultJwtPath)) {
+
+  // If jwtFilePath is provided, it should exist
+  if (typeof jwtFilePath === 'string' && !existsSync(jwtFilePath)) {
+    throw new Error(`No file exists at provided jwt secret path=${jwtFilePath}`)
+  }
+
+  if (typeof jwtFilePath === 'string' || existsSync(defaultJwtPath)) {
```

Comment:
Can it `existsSync` in case that type of `jwtFilePath` is not a string?

Reply:
I might be misunderstanding your comment, but the `existsSync` condition refers to the `defaultJwtPath`, while the first just checks if the user has provided a path. So this basically checks "has the user provided a path? if not, is there a file at the defaultJwtPath?"